bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikBaseThingHandler.java

```diff
@@ -0,0 +1,226 @@
+                .build();
+        if (!statusInfo.equals(getThing().getStatusInfo())) {
+            super.updateStatus(status, statusDetail, description);
+        }
+    }
+
+    private void scheduleRefreshJob() {
+        synchronized (this) {
+            if (refreshJob == null) {
+                int refreshPeriod = getVerifiedBridgeHandler().getBridgeConfig().refresh;
+                logger.debug("Scheduling refresh job every {}s", refreshPeriod);
+                refreshJob = scheduler.scheduleWithFixedDelay(this::scheduledRun, refreshPeriod, refreshPeriod,
+                        TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    private void cancelRefreshJob() {
+        synchronized (this) {
+            if (refreshJob != null) {
+                logger.debug("Cancelling refresh job");
+                refreshJob.cancel(true);
+                refreshJob = null;
+            }
+        }
+    }
+
+    private void scheduledRun() {
+        logger.trace("scheduledRun() called for {}", getThing().getUID());
+        try {
+            if (getVerifiedBridgeHandler() == null) {
+                updateStatus(OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Failed reaching out to RouterOS bridge");
+                return;
+            }
+            if (getBridge() != null && getBridge().getStatus() == OFFLINE) {
+                updateStatus(OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "The RouterOS bridge is currently offline");
+                return;
+            }
+
+            if (getThing().getStatus() != ONLINE)
+                updateStatus(ONLINE);
+            logger.debug("Refreshing all {} channels", getThing().getUID());
+            for (Channel channel : getThing().getChannels()) {
+                refreshChannel(channel.getUID());
+            }
+        } catch (Exception e) {
+            logger.warn("Unhandled exception while refreshing the {} Mikrotik thing", getThing().getUID(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    protected final void refresh() throws ChannelUpdateException {
+        if (getThing().getStatus() == ONLINE) {
+            if (getRouteros() != null) {
+                throttledRefreshModels();
+                for (Channel channel : getThing().getChannels()) {
+                    ChannelUID channelUID = channel.getUID();
+                    try {
+                        refreshChannel(channelUID);
+                    } catch (RuntimeException e) {
+                        throw new ChannelUpdateException(getThing().getUID(), channelUID, e);
+                    }
+                }
+            }
+        }
+    }
+
+    protected void throttledRefreshModels() {
+        MikrotikRouterosBridgeHandler bridgeHandler = (MikrotikRouterosBridgeHandler) getBridge().getHandler();
+        if (LocalDateTime.now().isAfter(lastModelsRefresh.plusSeconds(bridgeHandler.getBridgeConfig().refresh))) {
```

Comment:
Did you take a look into `ExpiringCache`? That would make this code obsolete.

Reply:
Great suggestion, thanks!